### PR TITLE
[core] [easy] Minor cleanup on protobuf

### PR DESCRIPTION
**Pull request by dentiny**
Two changes:
- Remove unused protobuf inclusion
- Remove `cc_enable_arenas`, which is deprecated by protobuf

Cleanup unused inclusion by compiler warning:
```
INFO: From Generating proto_library //src/ray/protobuf:export_driver_job_event_proto:
src/ray/protobuf/export_api/export_driver_job_event.proto:20:1: warning: Import src/ray/protobuf/common.proto is unused.
```

Checkout protobuf release notes for v3.14.0: https://github.com/protocolbuffers/protobuf/releases/tag/v3.14.0
```
Arenas are now unconditionally enabled. cc_enable_arenas no longer has
any effect.
```
Our protobuf version is v3.23.4
```
[~/protobuf] (main) 
ubuntu@hjiang-devbox-pg$ git show 2c5fa078d8e86e5f4bd34e6f4c9ea9e8d7d4d44a
commit 2c5fa078d8e86e5f4bd34e6f4c9ea9e8d7d4d44a (tag: v4.23.4, tag: v3.23.4, tag: v23.4)
Author: Protobuf Team Bot <protobuf-team-bot@google.com>
Date:   Wed Jul 5 14:53:30 2023 -0700

    Updating version.json and repo version numbers to: 23.4
```

### CODE REVIEW

File: bazel/ray_deps_setup.bzl [hunk 1]
```diff
@@ -92,7 +92,6 @@ def ray_deps_setup():
         sha256 = "76a33e2136f23971ce46c72fd697cd94dc9f73d56ab23b753c3e16854c90ddfd",
         strip_prefix = "protobuf-2c5fa078d8e86e5f4bd34e6f4c9ea9e8d7d4d44a",
         urls = [
-            # https://github.com/protocolbuffers/protobuf/commits/v23.4
```
Comment (dayshah) [inline bazel/ray_deps_setup.bzl:95]: why removing this link?

**Review by dayshah — APPROVED** [review]

File: bazel/ray_deps_setup.bzl (hunk 1 above)
Comment (dentiny) [reply bazel/ray_deps_setup.bzl:95]: It's not used, I'm against keep unused / commented-out code in our codebase for no reason.

Comment (jjyao) [reply bazel/ray_deps_setup.bzl:95]: This is a useful comment (not code) that tells us the protobuf version we are using.

Comment (dentiny) [reply bazel/ray_deps_setup.bzl:95]: I added another comment for version.

**Review by jjyao — APPROVED** [review]

**Pull request merged**